subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/OptionReader.java

```diff
@@ -82,14 +83,14 @@ public List<OptionDescriptor> getOptions(Object target, boolean validOnly) {
             JavaMethod<Object, Collection> optionValueMethod = cachedOptionValueMethods.get(optionElement);
             options.put(optionElement.getOptionName(), new InstanceOptionDescriptor(target, optionElement, optionValueMethod));
         }
+        options.putAll(CommandLineTaskConfigurer.getOppositeOptions(options.values(), target));
```

Comment:
:x: This adds a cyclic dependency between `OptionReader` and `CommandLineTaskConfigurer`!

It would be best to introduce a new type between `CommandLineTaskConfigurer` and `OptionReader`, keep `OptionReader` about just reading options from task classes and `CommandLineTaskConfigurer` about just configuring the task from the command line. The newly introduced type would be responsible for handling both built-in task options (`--rerun`) and boolean opposite options, it would be using `OptionReader` under the hood and be used by both `CommandLineTaskConfigurer` and the `Help` task`.

Reply:
Argh, true! Thanks for the clarification, I updated this accordingly.